source/components/temp_doeclim_component.cpp

```diff
@@ -0,0 +1,528 @@
+        } else if( varName == D_DIFFUSIVITY ) {
+            H_ASSERT( data.date == Core::undefinedIndex(), "date not allowed" );
+            diff = unitval::parse_unitval( data.value_str, data.units_str, U_CM2_S );
+	} else if( varName == D_AERO_SCALE ) {
+            H_ASSERT( data.date == Core::undefinedIndex(), "date not allowed" );
+            alpha = unitval::parse_unitval( data.value_str, data.units_str, U_UNITLESS );
+        } else if( varName == D_TGAV_CONSTRAIN ) {
+            H_ASSERT( data.date != Core::undefinedIndex(), "date required" );
+            tgav_constrain.set( data.date, unitval::parse_unitval( data.value_str, data.units_str, U_DEGC ) );
+        } else {
+            H_THROW( "Unknown variable name while parsing " + getComponentName() + ": "
+                    + varName );
+        }
+    } catch( bad_lexical_cast& castException ) {
+        H_THROW( "Could not convert var: "+varName+", value: " + data.value_str + ", exception: "
+                +castException.what() );
+    } catch( h_exception& parseException ) {
+        H_RETHROW( parseException, "Could not parse var: "+varName );
+    }
+}
+
+//------------------------------------------------------------------------------
+// documentation is inherited
+// TO DO: should we put these in the ini file instead?
+void TempDOECLIMComponent::prepareToRun() throw ( h_exception ) {
+    
+    H_LOG( logger, Logger::DEBUG ) << "prepareToRun " << std::endl;
+    
+    if( tgav_constrain.size() ) {
+        Logger& glog = Logger::getGlobalLogger();
+        H_LOG( glog, Logger::WARNING ) << "Temperature will be overwritten by user-supplied values!" << std::endl;
+    }
+    
+    dt = 1;         // years per timestep (this is hard-coded into Hector
+    ak = 0.31;      // slope in climate feedback - land-sea heat exchange linear relationship
+    bk = 1.59;      // offset in climate feedback - land-sea heat exchange linear relationship, W/m2/K
+    csw = 0.13;     // specific heat capacity of seawater W*yr/m3/K
+    earth_area = 5100656 * pow(10.0, 8);
+    kcon = 3155.0;  // conversion from cm2/s to m2/yr
+    q2co = 3.7;     // radiative forcing for atmospheric CO2 doubling, W/m2
+    rlam = 1.43;    // factor between land clim. sens. and sea surface clim. sens. T_L2x = rlam*T_S2x
+    secs_per_Year = 31556926.0;
+    zbot = 4000.0;  // bottom depth of diffusive ocean, m
+    bsi = 1.3;                  // warming factor for marine surface air over SST (due to retreating sea ice)
+    cal = 0.52;                 // heat capacity of land-troposphere system, W*yr/m2/K
+    cas = 7.80;                 // heat capacity of mixed layer-troposphere system, W*yr/m2/K
+    flnd = 0.29;                // fractional land area
+    fso = 0.95;                 // ocean fractional area below 60m
+    
+    
+    // Initializing all model components that depend on the number of timesteps (ns)
+    ns = core->getEndDate() - core->getStartDate() + 1;
```

Comment:
This is assuming that `dt` is 1, right?  Since you have defined `dt` above, you shouldn't be assuming anything about its value in the rest of the code.  Calculate `ns` using `dt` as if you didn't know anything about its value.